src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1106,7 +1143,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
             ext.beforeCheckout(this, build, git, listener);
         }
 
-        retrieveChanges(build, git, listener);
+        retrieveChanges(build, git, listener, GitUtils.workspaceToNode(workspace));
```

Comment:
That's how it was implemented in AbstractGitSCMSource.getCacheDir. There's a check for null Jenkins.getInstance. Now I'm using node to determine the cache directory to lock as well, so I need to have this check in my code. I've added it.